src/ci/docker/host-x86_64/x86_64-gnu-llvm-18/Dockerfile

```diff
@@ -50,6 +53,10 @@ ENV RUST_CONFIGURE_ARGS \
       --enable-llvm-link-shared \
       --set rust.thin-lto-import-instr-limit=10
 
-COPY host-x86_64/x86_64-gnu-llvm-16/script.sh /tmp/
+COPY host-x86_64/dist-x86_64-linux/shared.sh /scripts/
+COPY host-x86_64/dist-x86_64-linux/build-gccjit.sh /scripts/
+
+RUN /scripts/build-gccjit.sh /scripts
```

Comment:
Why these changes?

Reply:
That's syncing with changes that were made in 16 and 17, but not 18 yet -- I presume because they were developed around the same time 18 was added. But I think all of that is in question in #122546.